data/systemd/snapd.core-fixup.sh

```diff
@@ -11,6 +11,38 @@ if ! grep -q "ID=ubuntu-core" /etc/os-release; then
     exit 0
 fi
 
+# Workaround https://forum.snapcraft.io/t/5253
+#
+# We see sometimes corrupted uboot.env files created by fsck.vfat.
+# On the fat filesystem they are indistinguishable because one
+# has a fat16 name UBOOT.ENV (and not lfn (long-file-name)) but
+# the other has a "uboot.env" lfn name and a FSCK0000.000 FAT16
+# name. The only known workaround is to remove all dupes and put
+# one file back in place.
+if [ $(ls /boot/uboot | grep ^uboot.env$ | wc -l) -gt 1 ]; then
```

Comment:
This could be abbreviated to 
```
if [ $(ls /boot/uboot | grep -c ^uboot.env$) -gt 1 ]; then
```

Reply:
Ok, here's shellcheck:
```
In data/systemd/snapd.core-fixup.sh line 22:
if [ $(ls /boot/uboot | grep ^uboot.env$ | wc -l) -gt 1 ]; then
     ^-- SC2046: Quote this to prevent word splitting.
       ^-- SC2010: Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.
                        ^-- SC2126: Consider using grep -c instead of grep|wc -l.
```

Reply:
Good advice, I fix this in a followup. Except the "SC2010", as this is a bit atypical I'm not sure how to do the glob here in the way we need it.

Reply:
How about `find /boot/uboot -name uboot.env | wc -l`?